modules/properties-config/src/main/java/org/mule/runtime/config/internal/model/properties/ConfigurationPropertiesHierarchyBuilder.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.model.properties;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import org.mule.runtime.api.component.ConfigurationProperties;
+import org.mule.runtime.config.internal.dsl.model.config.CompositeConfigurationPropertiesProvider;
+import org.mule.runtime.config.internal.dsl.model.config.ConfigurationPropertiesResolver;
+import org.mule.runtime.config.internal.dsl.model.config.DefaultConfigurationPropertiesResolver;
+import org.mule.runtime.config.internal.dsl.model.config.DefaultConfigurationProperty;
+import org.mule.runtime.config.internal.dsl.model.config.EnvironmentPropertiesConfigurationProvider;
+import org.mule.runtime.config.internal.dsl.model.config.FileConfigurationPropertiesProvider;
+import org.mule.runtime.config.internal.dsl.model.config.GlobalPropertyConfigurationPropertiesProvider;
+import org.mule.runtime.config.internal.dsl.model.config.MapConfigurationPropertiesProvider;
+import org.mule.runtime.config.internal.dsl.model.config.SystemPropertiesConfigurationProvider;
+import org.mule.runtime.properties.api.ConfigurationPropertiesProvider;
+import org.mule.runtime.properties.api.ConfigurationProperty;
+import org.mule.runtime.properties.api.ResourceProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Stack;
+import java.util.function.Supplier;
+
+/**
+ * Builder of {@link ConfigurationPropertiesResolver} with the full hierarchy, considering the properties to be used. The
+ * hierarchy to consider is: 1. Deployment Properties 2. System Properties 3. Environment Properties 4. Application Properties
+ * (this includes file properties) 5. Global (Default) Properties This means that (for example) if we have a property 'hello' both
```

Comment:
line break missing?

Reply:
It was the formatter and me missing a stop 😅